gateway/src/apicast/policy/ngx_variable.lua

```diff
@@ -0,0 +1,15 @@
+local _M = {}
+
+local function context_values()
+ return {
+  uri = ngx.var.uri,
+  host = ngx.var.host,
+  remote_addr = ngx.var.remote_addr
+ }
+end
+
+function _M.available_context(policies_context)
+  return setmetatable(policies_context, { __index = context_values() })
```

Comment:
This still mutates the context. It should be the other way around. Setting the metatable of `context_values`.